CMakeLists.txt

```
@@ -0,0 +1,35 @@
+cmake_minimum_required(VERSION 3.10)
+
+# build options
+set(CMAKE_CXX_STANDARD 17)
+
+if (NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE Release)
+endif (NOT CMAKE_BUILD_TYPE)
+
+set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wpedantic -Werror -mtune=native -march=native")
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    message("\n##########################")
+    message("###### DEBUG BUILD #######")
+    message("##########################\n")
+else ()
+    message("\n############################")
+    message("###### RELEASE BUILD #######")
+    message("############################\n")
+endif ()
+
+# build
+project(server017 VERSION 0.0.1 DESCRIPTION "Server of team 17 for the Sopra")
+
+find_package(spdlog REQUIRED)
+
+file(GLOB_RECURSE SOURCES
+        ${CMAKE_SOURCE_DIR}/src/*.cpp
+        ${CMAKE_SOURCE_DIR}/src/*.hpp
+)
+
+SET(${LIBS} spdlog)
```

ich glaube bei mir musste ich das so mit "namespace" angeben, ich teste das gleich aber noch mal

```suggestion
SET(${LIBS} spdlog::spdlog CLI11::CLI11)
```

---

https://stackoverflow.com/a/48526017/4162386 

Habe das mal recherchiert:

> CMake will recognize that values passed to target_link_libraries() that contain :: in their name are supposed to be imported targets (rather than just library names), and will produce appropriate diagnostic messages if that target does not exist

Hört sich ganz gut an

---

Ah ja und das `stdc++fs` hab ich mit GCC 8.4 gebraucht...

---

Interessant, bei mir gings mit GCC 7.5 ohne, aber ich würde mal behaupten man geht im Zweifel auf die neue Version